### PR TITLE
Alternate startup logic to correctly wait on services to validate

### PR DESCRIPTION
**Pull request by omnunum**
SymlinkLibrary needed a validation step, and also needs to be moved to start after Symlinker to make sure the correct paths are created.  Additionally we should be waiting for all services to validate like we used to.

### CODE REVIEW

File: backend/program/libaries/symlink.py [hunk 1]
```diff
@@ -19,7 +19,21 @@ def __init__(self):
...
-
+        self.initialized = self.validate()
+        if not self.initialized:
+            logger.error("SymlinkLibrary initialization failed due to invalid configuration.")
+            return
+    
+    def validate(self) -> bool:
+        status = all(
+            os.path.exists(self.settings.library_path / d) 
+            for d in ( "shows", "movies", "anime_shows", "anime_movies")
```
Comment (dreulavelle) [inline backend/program/libaries/symlink.py:30]: You probably noticed by now, but we dont technically use the anime dir's just yet. I added a `is_anime` attr that we can use to populate those 2 dirs though.

Just gotta add sorting to this module to handle

Comment (omnunum) [reply backend/program/libaries/symlink.py:30]: We don't populate the library from them either at the moment

Comment (dreulavelle) [reply backend/program/libaries/symlink.py]: we can revisit it later when we're more stable.. but would like to sort anime out of the main 2 libraries.

File: backend/program/libaries/symlink.py [hunk 2]
```diff
@@ -19,7 +19,21 @@ def __init__(self):
...
+            logger.error("SymlinkLibrary initialization failed due to invalid configuration.")
+            return
+    
+    def validate(self) -> bool:
+        status = all(
+            os.path.exists(self.settings.library_path / d) 
+            for d in ("shows", "movies")
+        )
+        if not status:
+            folders = os.listdir(self.settings.library_path)
```
Comment (dreulavelle) [inline backend/program/libaries/symlink.py:33]: heh.. lol nice save

**Review by dreulavelle — APPROVED** [review]

**Pull request merged**